### PR TITLE
Fix #624 — Correctly rename archive folder when renaming list

### LINKED ISSUE #624: Archive not found when list renamed with capital letter
Version
-------

6.2.40

Installation method
-------------------
source package

Expected behavior
-----------------
When you rename a list with a capital letter, archive must match with list name.

Actual behavior
---------------
When you rename a list with a capital letter ( `Spf@lists.tld` for isntance) archives are renamed `Spf@lists.tld` while Sympa is looking for `spf@lists.tld` to display archives: it doesn't match and you can’t consult your archives.

If the list get mails, a new archive folder is created, all lowercase. You end with two archive folders:  `Spf@lists.tld` (from the renaming, can’t be used by Sympa) and `spf@lists.tld` (from the new mails).

Additional information
----------------------
I hope it's understandable :thinking:

### PR DESCRIPTION
**Pull request by ldidry**

### CODE REVIEW

**Comment by ikedas** [comment]
This may fix #624

File: src/lib/Sympa/Request/Handler/move_list.pm [hunk 1]
```diff
@@ -392,7 +392,7 @@ sub _move {
     # Try renaming archive.
     # Continue even if there are some troubles.
     my $arc_dir     = $current_list->get_archive_dir;
-    my $new_arc_dir = $fake_list->get_archive_dir;
+    my $new_arc_dir = lc($fake_list->get_archive_dir);
```
Comment (ikedas) [inline src/lib/Sympa/Request/Handler/move_list.pm]: Is this fix means `$fake_list` contains the list name with uppercase letter? I'll investigate it for a while.

**Comment by ldidry** [comment]
Yeah, sorry, I always forget to put the issue number in the description of the PR: I’m more used to Gitlab than Github and Gitlab can find the reference in the merge request subject.

File: src/lib/Sympa/Request/Handler/move_list.pm (hunk 1 above)
Comment (ldidry) [reply src/lib/Sympa/Request/Handler/move_list.pm]: Yes, it means that.

Comment (ikedas) [reply src/lib/Sympa/Request/Handler/move_list.pm]: Making line 187 in beginning of `_move()` to be
```perl
    my $listname     = lc($request->{listname} || );
```
looks better.

Comment (ikedas) [reply src/lib/Sympa/Request/Handler/move_list.pm]: Also line 459.

Comment (ldidry) [reply src/lib/Sympa/Request/Handler/move_list.pm]: I applied your suggestion and rewrote my commit (no need to keep my old solution in history).

**Pull request closed without merging**